projects/map-cesium/src/lib/map-cesium.service.ts

```diff
@@ -620,28 +620,40 @@ export class MapCesiumService {
 
   private create_geojson_layer(l: VectorLayer): GeoJsonDataSource {
     const newGeoJsonDataSource = new GeoJsonDataSource();
-    // default GeoJsonDataSource values
-    let fillColor = GeoJsonDataSource.fill;
-    let strokeColor = GeoJsonDataSource.stroke;
-    let strokeWidth = GeoJsonDataSource.strokeWidth;
+    // default UKIS values
+    let fillColor = Color.fromCssColorString('#FFFFFF99');
+    let strokeColor = Color.fromCssColorString('#3399CC');
+    let strokeWidth = 1;
+    let strokeOpacity = 1;
+    let fillOpacity = 1;
     let clamp = false;
-    if(l.options){
-      if(l.options['fillColor']){
-        fillColor = Object.freeze(Color.fromCssColorString(l.options['fillColor']));
-      }
-      if(l.options['strokeColor']){
-        strokeColor = Object.freeze(Color.fromCssColorString(l.options['strokeColor']));
-      }
-      if(l.options['strokeWidth']){
-        strokeWidth = l.options['strokeWidth'];
+
+    if(l.options && l.options.style){
+      const styleProperties = l.options.style(l.data)[0];
+      if(styleProperties){
+      fillColor = Color.fromCssColorString(styleProperties.fill_.color_) || fillColor;
```

Comment:
It is not optimal to use private properties from OpenLayers here, but we can change that later. 

Maybe we can define some common style properties. 
```ts
l.options.style{
 fillColor?: string;
 strokeColor?: string;
 strokeWidth?: string;
 strokeOpacity?: number;
 fillOpacity?: number;
 circleRadius?: number;
 iconImg?: any;
 ...
}
```

@voinSR what do you think about it?

Reply:
This would be ideal.